### PR TITLE
client: add basic cli test

### PR DESCRIPTION
**Pull request by ryanio**
This PR adds a basic cli test that spins up the client, imports a block from a peer, and finishes.

The Github Action is specified with a matrix of `network` and `syncmode`.

### CODE REVIEW

**Comment by holgerd77** [comment]
Test is failing here so this is `WIP` I guess? 🤔

**Comment by holgerd77** [comment]
Rebased this.

**Comment by ryanio** [comment]
Screen Shot 2021-07-08 at 6 02 24 PM

this is ready for review! I suggest to start by not making these new checks required so we can use it as extra sanity checks that all the monorepo package components can be built into a functioning client.

**Review by acolytec3 — APPROVED** [review]
Nice!  I was able to pass the tests on rinkeby with full sync though had trouble connecting to either with LES (kind of like you described).  Does it make sense to disable libp2p transport since we're not going to be running any other libp2p nodes for this test just as a way to slightly minimize resource usage?  I realize that's a small thing but just a thought.  Can be merged either way though!

File: .github/workflows/client-build.yml [hunk 1]
```diff
@@ -56,3 +56,37 @@ jobs:
...
+      # Bootstraps the current package only
+      - run: npm install --ignore-scripts && npm run bootstrap:client
+        if: steps.cache.outputs.cache-hit != 'true'
+        working-directory: ${{github.workspace}}
+
+      # Builds current package and the ones it depends from.
+      - run: npm run build:client
+        working-directory: ${{github.workspace}}
+
+      - run: npm run test:cli -- --network=${{matrix.network}} --syncmode=${{matrix.syncmode}}
```
Comment (acolytec3) [inline .github/workflows/client-build.yml:92]:
```suggestion
      - run: npm run test:cli -- --network=${{matrix.network}} --syncmode=${{matrix.syncmode}} --transports rlpx
```

Comment (ryanio) [reply .github/workflows/client-build.yml:92]: yes for sure, nice suggestion

**Review by holgerd77 — APPROVED** [review]
Nice, LGTM, thanks Ryan! I guess this can be merged?

**Pull request merged**